packages/hashtower.sol/contracts/HashTower.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {PoseidonT3} from "poseidon-solidity/PoseidonT3.sol";
+
+// CAPACITY = W * (W**0 + W**1 + ... + W**(H - 1)) = W * (W**H - 1) / (W - 1)
+uint256 constant H = 24;
+uint256 constant W = 4;
+uint256 constant CAPACITY = 375_299_968_947_540;
```

Comment:
These should not be constants and should be set by the user. Definitely allow a variable `H`, _maybe_ allow a variable `W` within a range (maybe like `[2-6]`).

Reply:
I think we may choose a large enough `H` in the contract for the library users without incurring costs:
- the storage slots at the high levels will remain 0 before being used
- library users can pick their own `H` in the circuit independent of the `H` in the contract
- as the count of items growths, library users can upgrade the circuit without changing the storage

Reply:
For `W`, it does lead to a tradeoff between the circuit complexity and the on-chain cost.
Increasing W will linearly increase the circuit complexity, and decrease the on-chain cost a little bit.
I have done an outdated calculation before. I should do it for the new contract / circuit.

I think there are some possibilities:

1. Fixed W
1.1 Choose a fixed W for the library users
1.2 Deploy different copies for W=[2-6]
2. Dynamic W
2.1 Merge W with countOfItems, so it will not incur an additional SLOAD

Reply:
For `W`, it does lead to a tradeoff between the circuit complexity and the on-chain cost.
Increasing W will linearly increase the circuit complexity, and decrease the on-chain cost a little bit.
I have done an outdated calculation before. I should do it for the new contract / circuit.

I think there are some possibilities:

1. Static W
1.1 Choose a fixed W for the library users
1.2 Deploy different copies for W=[2-6]
2. Dynamic W
2.1 Merge W with countOfItems, so it will not incur an additional SLOAD